server/bundler/loader.js

```diff
@@ -101,8 +112,38 @@ function splitTemplate( path, section ) {
 	return result.join( '\n' );
 }
 
-function requireTemplate( module ) {
-	return 'require( ' + JSON.stringify( module ) + ' )( controller.clientRouter );\n';
+function getPathRegex( pathString ) {
+	var regex;
+
+	if ( pathString === '/' ) {
+		return JSON.stringify( pathString );
+	}
+
+	regex = utils.pathToRegExp( pathString );
+
+	return '/' + regex.toString().slice( 1, -1 ) + '/';
+}
+
+function requireTemplate( section ) {
+	var pathRegex,
+		result = [];
+
+	section.paths.forEach( function( path ) {
+		pathRegex = getPathRegex( path );
+
+		result = result.concat( [
```

Comment:
You're actually reducing here, so you could use [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce) instead of the `forEach` on L131, and modify this part accordingly.